logisland-components/logisland-processors/logisland-processor-web-analytics/src/main/java/com/hurence/logisland/processor/webanalytics/SetSourceOfTraffic.java

```diff
@@ -346,19 +346,26 @@ private boolean adwords(final String location, final SourceOfTrafficMap sourceOf
             sourceOfTraffic.setKeyword(adwords ? ADWORDS : DOUBLECLICK);
 
             try {
-                String[] params = new URI(location).getQuery().split("&");
-                for(String param: params) {
-                    String[] token = param.split("campaign=");
-                    if ( token.length==2) {
-                        sourceOfTraffic.setCampaign(token[1]);
-                    }
-                }
+                setAdwordsCampaign(new URI(location).getQuery().split("&"), sourceOfTraffic);
             } catch (URISyntaxException e) {
-                getLogger().error("URISyntaxException", e);
+                // Symbol `|` generates exception in URI but not in URL.
+                try {
+                    setAdwordsCampaign(new URL(location).getQuery().split("&"), sourceOfTraffic);
+                } catch (final MalformedURLException e2) {
```

Comment:
The method getQuery of a URL may be different of an URI so you should test with an exemple in test. (For exemple with one value that we get an exception in our clients)

Reply:
Added tests with '|' that break URL.